tasks/qemu.yml

```diff
@@ -6,37 +6,61 @@ vars:
     /usr/share/OVMF/OVMF_CODE.fd
     /usr/share/edk2/ovmf/OVMF_CODE.fd
     /usr/share/edk2-ovmf/x64/OVMF_CODE.fd
+  OVMF_CODE:
+    sh: |
+      for f in {{.OVMF_LOCATIONS}}; do
+      if [ -f "$f" ]; then
+      echo $f
+      break
+      fi
+      done
+  OVMF_VARS_DEFAULT:
+    sh: echo {{.OVMF_CODE}} | sed 's/CODE/VARS/g'
+  OVMF_VARS: out/artifacts/OVMF_VARS.fd
 
 tasks:
+
+  clean-efivars:
+    desc: "Clean EFI variables"
+    cmds:
+      - cmd: mkdir -p $(dirname {{.OVMF_VARS}})
+        silent: true
+      - cp {{.OVMF_VARS_DEFAULT}} {{.OVMF_VARS}}
+
+  efivars:
+    cmds:
+      - cmd: mkdir -p $(dirname {{.OVMF_VARS}})
+        silent: true
+      - cp {{.OVMF_VARS_DEFAULT}} {{.OVMF_VARS}}
+    status:
+      - '[ -f {{.OVMF_VARS}} ]'
+
   # internal use only
   qemu:
     deps:
       - task: :deps:check-cmd
         vars: {CMD: "qemu-system-x86_64"}
+      - efivars
```

Comment:
Is that syntax correct or should it be `- task: efivars`?

Reply:
Task has multiple ways to unmarshal tasks. If it is a string, it will be evaluated as task name.
If you need to adapt some fields like `silent` or `vars`, you have to parse task as object.

Reply:
Thanks for clarifying!